src/widgets/CredentialsBox.tsx

```diff
@@ -50,13 +61,17 @@ export class GitCredentialsForm
   /**
    * Returns the input value.
    */
-  getValue(): Git.IAuth {
+  getValue(): Git.IAuthWithCacheCredentials {
     return {
       username: this._user.value,
-      password: this._password.value
+      password: this._password.value,
+      cacheCredentials: this._checkboxCacheCredentials.checked
     };
```

Comment:
I would extend `Git.IAuth` with a new optional parameter `cacheCredentials`. This will allow you to simply pass-by those information through the model and then to the server. One reason for that is the strong bound between that parameter and the credentials; passing it as a separate argument as done currently would be interesting if it had an effect even if the authentication was not defined.